### ISSUE TITLE
install --recursive could run jobs in parallel

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#450]
created a new label `do-in-parallel` to collect use-cases which beg for parallelization, and seems to be trivial to be implemented

specific implementation approach yet to be discussed, thus tagged for conference agenda

``` shell
$> datalad install --recursive 1 -l info --source http://datasets.datalad.org/
2016-05-20 19:01:20,681 [INFO   ] Creating a new git repo at /mnt/datasets/datalad/test-crawled/datasets.datalad.org (install.py:634)
2016-05-20 19:01:20,715 [ERROR  ] 'git clone -v http://datasets.datalad.org/ /mnt/datasets/datalad/test-crawled/datasets.datalad.org' returned with exit code 128
| stderr: 'Cloning into '/mnt/datasets/datalad/test-crawled/datasets.datalad.org'...
| fatal: repository 'http://datasets.datalad.org/' not found
| ' (gitrepo.py:321)
2016-05-20 19:01:20,716 [INFO   ] Creating a new git repo at /mnt/datasets/datalad/test-crawled/datasets.datalad.org (install.py:634)
2016-05-20 19:01:20,840 [INFO   ] Creating a new git repo at /mnt/datasets/datalad/test-crawled/datasets.datalad.org/openfmri (install.py:634)
2016-05-20 19:01:20,986 [INFO   ] Creating a new git repo at /mnt/datasets/datalad/test-crawled/datasets.datalad.org/openfmri/ds000011 (install.py:634)
...
```

Since we are cloning them manually first, and then only 'registering' as a submodule under that path, it sounds like it should be trivial to implement in parallel -- those are independent repositories to clone.

**Comment by yarikoptic** [comment]
related complication:  inability to serialize GitPython's Git -- https://github.com/gitpython-developers/GitPython/issues/504

**Comment by mih** [comment]
Many things became simpler in this regard. However `install/get` still do not support this. With http://handbook.datalad.org/en/latest/basics/101-146-gists.html#parallelize-subdataset-processing the pressure to have built-in parallelization is much reduced.
However, that paradigm is not applicable to subdataset installation. I tried parallel installation of 42k subdatasets via:

```
datalad -f '{path}' subdatasets | xargs -n 1 -P10 datalad get -n
```

after I estimated a standard serial run `datalad get -n -r` to take somewhere between 25h and 40h. It finished after 18h with parallelization (not a stellar result). However, it left the dataset in a broken state due to random locking issues of this kind:

```
CommandError: 'git submodule update --checkout --init -- sub-6025317' failed with exitcode 128 under /data/project/ukb_vbm/bidsclone                                                      
error: could not lock config file .git/config: File exists                                  
fatal: Failed to register url for submodule path 'sub-6025317'                               
```

I think there is a solid potential for speed-up following the direction of https://github.com/datalad/datalad/pull/4793 and fully bypass `git-submodule`, and replace it with `datalad clone` plus `git config` calls. The former will be independent across subdatasets. The latter we can protect with a locking mechanism to avoid concurrent execution.
